### ISSUE TITLE
`bind` data not available in run or init hooks

### ISSUE THREAD

**Issue by smith** [habitat-sh/habitat#1489]
When starting a service with: `hab start myorigin/mysvc --bind elasticsearch:elasticsearch.default`, with a run or init hook that contains:

```
echo "{{toml bind}}"
```

it will output:

```
has_elasticsearch = false
```

if you make a health_check hook that contains `echo "{{toml bind}}"`, then hit it immediately after  the service starts, you get:

```
has_elasticsearch = false
```

but if you wait a few seconds you get:

```
has_elasticsearch = true

# all the rest of the toml you would expect
```

So, it looks like the binding stuff is happening some time after the run hook has run, so I'm not able to use it in my run hook.

Is this something I should expect to work or is there another way to use the binding data in a run hook or is that something I shouldn't be doing?

**Comment by adamhjk** [comment]
The new sup will render this correctly eventually, unlike the old one. What it won't solve is the timing question. 

I suspect we need some sort of precondition to starting. Let's noodle more on options.

**Comment by elliott-davis** [comment]
I also noticed this behavior. The way I worked around it was supplying a config file that called `has_<my_binding>`. From then on everything seemed to work in the correct order for the run hook. I haven't tested with the init hook.

**Comment by reset** [comment]
@adamhjk and I chatted right after #1667 landed about this issue. What we're going to do is make the bind settings part of what a package expects and check this at build and runtime. You'll set something like `pkg_binds_required=([database]=(port host ))` which will allow you to set a bind named `database` which *must* bind to a service group which hosts a service that has the config variable `port` and `host` exported.

With this, we can also make sure that you wait until your bind is available before starting. This is the timing bit that @adamhjk was mentioning in his reply in November. When we setup a bind to a service group we wait until that service group becomes available and, when it does, it should be exporting the variables that we're expecting. If it doesn't, then we warn the user and don't start the process.

**Comment by smurawski** [comment]
Part of this was added in #1667 and part of the development is being planned.

**Comment by reset** [comment]
This is fixed by #2076 and the multi-tenant supervisor work, thanks for the report!
